### ISSUE TITLE
[Feature Request]: adjust cpu and memory request and limit sizes in manifests

### ISSUE THREAD

**Issue by shalberd** [opendatahub-io/odh-dashboard#1786]
### Feature description

Currently, cpu and memory resource settings for odh dashboard are in no way representative of real usage.

https://github.com/opendatahub-io/odh-dashboard/blob/main/manifests/base/deployment.yaml#L34

https://github.com/opendatahub-io/odh-dashboard/blob/main/manifests/base/deployment.yaml#L34

![Bildschirmfoto 2023-09-14 um 12 55 41](https://github.com/opendatahub-io/odh-dashboard/assets/21118431/6344e56b-9b05-453a-8bb9-e340fd81a10d)

![Bildschirmfoto 2023-09-14 um 13 01 06](https://github.com/opendatahub-io/odh-dashboard/assets/21118431/81ad8837-8a0b-43c6-9683-df180603f07c)

![Bildschirmfoto 2023-09-14 um 13 01 34](https://github.com/opendatahub-io/odh-dashboard/assets/21118431/a0c1ec28-0c56-4f1b-9f67-d0a05cb4f0ec)

Let's start with CPU:

there is no reason for the pod to only be scheduled if 0.5 CPU is free on any worker node. 
A request of 200m would be more than enough. Keep in mind: We've got two pods of odh-dashboard running, which means in case we change this setting, there is already 0.3 * 2 = 0.6 CPU Cores available for scheduling other containers and pods on the node.

Average usage in my experience never even exceeded 200m, so the limit can be set to let's say 250m, just in case.

Regarding memory:

We have a relatively simple Web app here ... average memory usage in my monitoring never exceeded 350Mi.

So, I'd say requests.memory should be changed from 1Gi to 350Mi and 
limits.memory should be changed from 2Gi to 400Mi. 

Using these settings, in terms of QoS, we allow for some bursting, while not over-reserving (requests) too much initially.

What do you think?

Thank you, @tkolo for pointing me to this. https://github.com/opendatahub-io/odh-dashboard/issues/1730#issuecomment-1718122083

to summarize, I'd propose changing the deployment.yaml section to the following for odh-dashboard

```
        resources:
          limits:
            cpu: 300m
            memory: 350Mi
          requests:
            cpu: 200m
            memory: 250Mi
```

and for oauth-proxy:

```
        resources:
          limits:
            cpu: 100m
            memory: 120Mi
          requests:
            cpu: 50m
            memory: 100Mi
```

**Comment by shalberd** [comment]
tested on our cluster and working well for us, with about 5-10 dashboard users.
